internal/wasm/jit/asm/amd64/impl.go

```diff
@@ -0,0 +1,401 @@
+func (a *assemblerImpl) CompileJumpToRegister(jmpInstruction asm.Instruction, reg asm.Register) {
+	n := a.newNode(jmpInstruction, operandTypesNoneToRegister)
+	n.dstReg = reg
+}
+
+// CompileReadInstructionAddress implements asm.AssemblerBase.CompileReadInstructionAddress
+func (a *assemblerImpl) CompileReadInstructionAddress(destinationRegister asm.Register, beforeAcquisitionTargetInstruction asm.Instruction) {
+	// TODO
+}
+
+// CompileRegisterToRegisterWithMode implements assembler.CompileRegisterToRegisterWithMode
+func (a *assemblerImpl) CompileRegisterToRegisterWithMode(instruction asm.Instruction, from, to asm.Register, mode int64) {
+	n := a.newNode(instruction, operandTypesRegisterToRegister)
+	n.srcReg = from
+	n.dstReg = to
+	n.mode = byte(mode)
+}
+
+// CompileMemoryWithIndexToRegister implements assembler.CompileMemoryWithIndexToRegister
+func (a *assemblerImpl) CompileMemoryWithIndexToRegister(instruction asm.Instruction, srcBaseReg asm.Register, srcOffsetConst int64, srcIndex asm.Register, srcScale int16, dstReg asm.Register) {
+	n := a.newNode(instruction, operandTypesMemoryToRegister)
+	n.srcReg = srcBaseReg
+	n.srcConst = srcOffsetConst
+	n.srcMemIndex = srcIndex
+	n.srcMemScale = byte(srcScale)
+	n.dstReg = dstReg
+}
+
+// CompileRegisterToMemoryWithIndex implements assembler.CompileRegisterToMemoryWithIndex
+func (a *assemblerImpl) CompileRegisterToMemoryWithIndex(instruction asm.Instruction, srcReg asm.Register, dstBaseReg asm.Register, dstOffsetConst int64, dstIndex asm.Register, dstScale int16) {
+	n := a.newNode(instruction, operandTypesRegisterToMemory)
+	n.srcReg = srcReg
+	n.dstReg = dstBaseReg
+	n.dstConst = dstOffsetConst
+	n.dstMemIndex = dstIndex
+	n.dstMemScale = byte(dstScale)
+}
+
+// CompileRegisterToConst implements assembler.CompileRegisterToConst
+func (a *assemblerImpl) CompileRegisterToConst(instruction asm.Instruction, srcRegister asm.Register, value int64) asm.Node {
+	n := a.newNode(instruction, operandTypesRegisterToConst)
+	n.srcReg = srcRegister
+	n.dstConst = value
+	return n
+}
+
+// CompileRegisterToNone implements assembler.CompileRegisterToNone
+func (a *assemblerImpl) CompileRegisterToNone(instruction asm.Instruction, register asm.Register) {
+	n := a.newNode(instruction, operandTypesRegisterToNone)
+	n.srcReg = register
+}
+
+// CompileNoneToRegister implements assembler.CompileNoneToRegister
+func (a *assemblerImpl) CompileNoneToRegister(instruction asm.Instruction, register asm.Register) {
+	n := a.newNode(instruction, operandTypesNoneToRegister)
+	n.dstReg = register
+}
+
+// CompileNoneToMemory implements assembler.CompileNoneToMemory
+func (a *assemblerImpl) CompileNoneToMemory(instruction asm.Instruction, baseReg asm.Register, offset int64) {
+	n := a.newNode(instruction, operandTypesNoneToMemory)
+	n.dstReg = baseReg
+	n.dstConst = offset
+}
+
+func (a *assemblerImpl) CompileConstToMemory(instruction asm.Instruction, value int64, dstbaseReg asm.Register, dstOffset int64) asm.Node {
```

Comment:
maybe this and below are missing the docs that say it implements the interface, or if it doesn't document them here 🙏